content/docs/esc/other-integrations/github-actions.md

```diff
@@ -0,0 +1,219 @@
+{{< /notes >}}
+
+In addition to the Pulumi ESC and GitHub Actions integrations below, consider authenticating to your Pulumi Cloud with GitHub OIDC. [Follow the guide](https://www.pulumi.com/docs/pulumi-cloud/oidc/client/github/) to enable the [Pulumi Auth Action](https://github.com/marketplace/actions/pulumi-auth-action).
+
+## Log in to your GitHub account
+
+ESC enables you to log into your GitHub account using credentials stored in an ESC Environment. This allows you to programmatically log in to your GitHub account from your terminal without having to locally store your personal access token.
+
+Note, that a fine-grained GitHub personal access token is required. Follow the [GitHub instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) to create one.
+
+### Create an ESC Environment
+
+Use the Pulumi ESC CLI to create and configure an Environment. Alternatively, follow the [Pulumi Cloud console instructions](https://www.pulumi.com/docs/esc/get-started/create-environment/#create-via-the-console).
+
+```bash
+# ensure you're logged in
+$ esc login
+# Logged in to pulumi.com as ...
+
+# create a new ESC Environment
+$ ESC_ENV=my-gh-login
+$ esc env init ${ESC_ENV}
+# Environment created.
+```
+
+Add your fine-grained GitHub personal access token to the Environment.
+
+```bash
+# Replace github_pat_123abc with your valid token
+$ esc env set ${ESC_ENV} files.GH_PAT github_pat_123abc --secret
+```
+
+Note: the GitHub token is declared as a `secret`. Once the Environment is saved, Pulumi will encrypt its value and replace it with ciphertext.
+
+Now that the Pulumi ESC Environment is created, it can be consumed in a variety of ways, such as running other shell commands without having to set the environment variables locally first.
+
+### Use ESC with `gh login`
+
+The `esc run` command opens the Environment you previously created, sets the specified environment variables into a temporary environment, and then uses those environment variables in the context of the `gh` commands.
+
+Log into your GitHub account without needing to manage the credentials locally.
+
+```bash
+# ensure you're currently not logged in
+$ gh auth logout
+
+# retrieve the esc environment, and
+# authenticate programmatically against github.com
+$ ESC_ENV=my-gh-login
+$ esc run -i ${ESC_ENV}  -- sh -c 'gh auth login --with-token < $GH_PAT'
+
+# confirm you're now logged in
+$ gh auth status
+# github.com
+#   ✓ Logged in to github.com account desteves (keyring)
+#   - Active account: true
+#   - Git operations protocol: https
+#   - Token: github_pat_11AAW2NYI0gtUYVcFfShHL_***
+```
+
+For additional options and details, see `esc run --help`.
+
+Using Pulumi Infrastructure as Code (IaC) with GitHub Actions? The next section demonstrates Pulumi ESC alongside Pulumi IaC deployments.
```

Comment:
I'm a little confused about the focus of this page. The title implies that this is about integrating ESC with Github Actions - but these sections don't have anything to do with that - they appear to be about using ESC to log in to the `gh` CLI?

Reply:
Hey Komal, thank you for looking at this. There is a [separate issue](https://github.com/pulumi/docs/issues/12503) to track additional ESC with GitHub Actions use cases on this page. The initial scope is to cover the gh cli example you mentioned as well as integrate our Pulumi GH Action + ESC. The later example is more on par with the page title. The goal is to publish these initial use cases for the recent GitHub + Pulumi workshop and add more cases after PulumiUp.